src/voxelalgorithms.cpp

```diff
@@ -618,9 +630,8 @@ void update_lighting_nodes(Map *map,
 			modified_blocks);
 		// Initialize light values for light spreading.
 		for (u8 i = 0; i <= LIGHT_SUN; i++) {
-			const std::vector<ChangingLight> &lights = light_sources.lights[i];
-			for (std::vector<ChangingLight>::const_iterator it = lights.begin();
-					it < lights.end(); ++it) {
+			const auto &lights = light_sources.lights[i];
+			for (auto it = lights.begin(); it < lights.end(); ++it) {
```

Comment:
Range-based `for` loop would be possible now. Same for the other occurrences.

Reply:
I opted to not touch this code too much because it looked complicated.